core/src/autogluon/core/scheduler/scheduler_constructor.py

```diff
@@ -0,0 +1,131 @@
+
+
+_scheduler_presets = {
+    'auto': {'searcher': 'random'},
+    # 'grid': {'searcher': 'grid'},  # grid commented out as it isn't compatible with most default model search spaces
+    'random': {'searcher': 'random'},
+    'bayesopt': {'searcher': 'bayesopt'},
+    # 'skopt': {'searcher': 'skopt'},  # TODO: Remove skopt? Is it worthwhile to keep as an option?
+    # Don't include hyperband and bayesopt hyperband at present
+}
+
+
+def init_scheduler(
+        hyperparameter_tune_kwargs,
+        time_out: float = None,
+        num_trials: int = None,
+        nthreads_per_trial='all',
+        ngpus_per_trial='all',
+        **kwargs):
+    """
+    Constructs a scheduler via lazy initialization based on the input hyperparameter_tune_kwargs.
+
+    Parameters
+    ----------
+    hyperparameter_tune_kwargs : str or dict
+        Hyperparameter tuning strategy and kwargs.
+        If None, then hyperparameter tuning will not be performed.
+        Valid preset values:
+            'auto': Uses the 'random' preset.
+            'random': Performs HPO via random search.
+            'bayesopt': Performs HPO via bayesian optimization.
+        For valid dictionary keys, refer to :class:`autogluon.core.scheduler.FIFOScheduler` documentation.
+            The 'searcher' key is required when providing a dict.
+    time_out : float, default = None
+        Same as hyperparameter_tune_kwargs['time_out']. Ignored if specified in hyperparameter_tune_kwargs.
+        At least one of time_out or num_trials must be specified.
+    num_trials : int, default = None
+        Same as hyperparameter_tune_kwargs['time_out']. Ignored if specified in hyperparameter_tune_kwargs.
+        At least one of time_out or num_trials must be specified.
+    nthreads_per_trial : str or int, default = 'all'
+        Same as hyperparameter_tune_kwargs['nthreads_per_trial']. Ignored if specified in hyperparameter_tune_kwargs.
+        Number of CPU threads to use per HPO trial.
+        Valid str values:
+            'all': Use all CPU threads.
+            'auto': Keep value as 'auto' in output, must be updated downstream.
+        If None, use all CPU threads as in 'all'.
+    ngpus_per_trial : str or int, default = 'all'
+        Same as hyperparameter_tune_kwargs['ngpus_per_trial']. Ignored if specified in hyperparameter_tune_kwargs.
+        Number of GPUs to use per HPO trial.
+        Valid str values:
+            'all': Use all GPUs.
+            'auto': Keep value as 'auto' in output, must be updated downstream.
+        If None, use 0 GPUs.
+    **kwargs :
+        Kwargs to specify any other scheduler parameters.
+        A kwarg will be ignored if also specified in hyperparameter_tune_kwargs.
+
+    Returns
+    -------
+    scheduler_options : Tuple of (scheduler_cls : class :class:`autogluon.core.scheduler.TaskScheduler`, scheduler_params : dict)
+        To construct the scheduler, call `scheduler_options[0](train_fn, **scheduler_options[1])`
```

Comment:
Maybe clearer: To actually construct a Scheduler object, call ...

Reply:
updated